master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -198,10 +215,23 @@ private[celeborn] class Master(
   }
   masterSource.addGauge(MasterSource.IS_ACTIVE_MASTER) { () => isMasterActive }
 
+  private val threadsStarted: AtomicBoolean = new AtomicBoolean(false)
```

Comment:
QQ: Is this an unrelated bugfix ?

Reply:
No, it is related to this change. The `onStart()` of the new `internalRpcEndpoint` calls the `onStart()` of the existing endpoint (so does the `onStop`). The `onStart`/`onStop` is scheduling some common tasks/cancelling it and we don't want it to be called multiple times for different endpoints.